src/Microsoft.SqlTools.ServiceLayer/EditData/EditSession.cs

```diff
@@ -203,13 +235,51 @@ public EditUpdateCellResult UpdateCell(long rowId, int columnId, string newValue
 
             // Attempt to get the row that is being edited, create a new update object if one
             // doesn't exist
-            RowEditBase editRow = EditCache.GetOrAdd(rowId, new RowUpdate(rowId, associatedResultSet, objectMetadata));
+            // NOTE: This *must* be done as a lambda. RowUpdate creation requires that the row
+            // exist in the result set. We only want a new RowUpdate to be created if the edit
+            // doesn't already exist in the cache
+            RowEditBase editRow = EditCache.GetOrAdd(rowId, key => new RowUpdate(rowId, associatedResultSet, objectMetadata));
 
             // Pass the call to the row update
             return editRow.SetCell(columnId, newValue);
         }
 
         #endregion
 
+        #region Private Helpers
```

Comment:
In a previous CR earlier, Kevin suggested not putting private regions if it's just one function. It made looking up private functions harder than usual.